twister missing build warning in cpp files

**Describe the bug**

Hey folks, bumped into some cpp tests that are generating warnings, specifically

```console
$ west build -p -b native_sim -T tests/subsys/rtio/rtio_i2c/rtio.i2c
...
zephyrproject/zephyr/include/zephyr/drivers/emul.h:117:49: warning: unnecessary parentheses in declaration of ‘__emulreg_DT_N_S_i2c_100_S_blocking_emul_80_bus’ [-Wparentheses]
...
$ west build -p -b native_sim -T tests/drivers/i2c/i2c_emul/drivers.i2c.emul.target_buf
...
/usr/local/google/home/fabiobaltieri/zephyrproject/zephyr/tests/drivers/i2c/i2c_emul/src/test_fowarding_common.cpp:75:40: warning: ‘data’ may be used uninitialized [-Wmaybe-uninitialized]
```

but the same do not seem to be caught by twister

```console
$ ./scripts/twister -v --testsuite-root tests/drivers/i2c/i2c_emul
...
INFO    - 1/2 native_sim                tests/drivers/i2c/i2c_emul/drivers.i2c.emul.target_buf PASSED (native 0.020s)
INFO    - 2/2 native_sim                tests/drivers/i2c/i2c_emul/drivers.i2c.emul.target_pio PASSED (native 0.020s)
...
$ ./scripts/twister -v --testsuite-root tests/subsys/rtio/rtio_i2c
...
INFO    - 1/1 native_sim                tests/subsys/rtio/rtio_i2c/rtio.i2c                PASSED (native 0.034s)
```

The error shows up build.log, shouldn't the twister run fail?

**To Reproduce**
The commands above.

**Expected behavior**
Twister is unhappy.

**Impact**
Confusion

**Environment (please complete the following information):**
Linux, Zephyr SDK 0.16.8, cc544803811

--- a/include/zephyr/drivers/emul.h
+++ b/include/zephyr/drivers/emul.h
@@ -114,7 +114,7 @@ struct emul {
  * the bus its on. Intended for use in other internal macros when declaring {bus}_emul
  * structs in peripheral emulators.
  */
-#define Z_EMUL_REG_BUS_IDENTIFIER(_dev_node_id) (_CONCAT(_CONCAT(__emulreg_, _dev_node_id), _bus))
+#define Z_EMUL_REG_BUS_IDENTIFIER(_dev_node_id) _CONCAT(_CONCAT(__emulreg_, _dev_node_id), _bus)
 
 /* Conditionally places text based on what bus _dev_node_id is on. */
 #define Z_EMUL_BUS(_dev_node_id, _i2c, _espi, _spi, _mspi, _uart, _none)                           \

--- a/tests/drivers/i2c/i2c_emul/src/test_fowarding_common.cpp
+++ b/tests/drivers/i2c/i2c_emul/src/test_fowarding_common.cpp
@@ -55,7 +55,7 @@ ZTEST_SUITE(i2c_emul_forwarding, NULL, i2c_emul_forwarding_setup, i2c_emul_forwa
 
 ZTEST(i2c_emul_forwarding, test_invalid_address_for_target)
 {
-	uint8_t data;
+	uint8_t data = 0;
 	int rc = i2c_write(targets[0], &data, 1, emulated_target_config[0].address + 1);
 	zassert_equal(-EINVAL, rc, "Expected %d (-EINVAL), but got %d", -EINVAL, rc);
 	zexpect_equal(0, target_read_requested_0_fake.call_count);
@@ -69,7 +69,7 @@ ZTEST(i2c_emul_forwarding, test_invalid_address_for_target)
 
 ZTEST(i2c_emul_forwarding, test_error_in_stop)
 {
-	uint8_t data;
+	uint8_t data = 0;
 
 	target_stop_0_fake.return_val = -EINTR;
 	zassert_equal(-EINTR, i2c_write(controller, &data, 1, emulated_target_config[0].address));

